pandas/tests/series/methods/test_append.py

```diff
@@ -61,6 +61,18 @@ def test_append_tuples(self):
 
         tm.assert_series_equal(expected, result)
 
+    def test_append_dataframe_raises(self):
+        # GH 30975
+        df = pd.DataFrame({"A": [1, 2], "B": [3, 4]})
+        li = [df.B, df]
+
+        msg = "to_append should be a Series or list/tuple of Series, got DataFrame"
+        with pytest.raises(TypeError, match=msg):
+            df.A.append(df)
+        msg = "to_append should be a Series or list/tuple of Series, got list"
+        with pytest.raises(TypeError, match=msg):
+            df.A.append(li)
```

Comment:
could use just the minimum to invoke TypeError
```suggestion
            df.A.append([df])
```